corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -1,23 +1,51 @@
 (function ($, _) {
-    var CaseProperty = function (data) {
-        this.name = ko.observable(data.name);
-        this.type = ko.observable(data.type);
-        this.description = ko.observable(data.description);
+    var CaseProperty = function (caseType, data, casePropertyUrl) {
+        var self = this;
+
+        self.caseType = ko.observable(caseType);
+        self.name = ko.observable(data.name);
+        self.type = ko.observable(data.type);
```

Comment:
I still think you should call this `dataType` as there are now 2 different `types` associated with the case